assets/js/blocks/price-filter/index.js

```diff
@@ -64,10 +64,30 @@ registerBlockType( 'woocommerce/price-filter', {
 			default: 3,
 		},
 	},
-
+	transforms: {
+		from: [
+			{
+				type: 'block',
+				blocks: [ 'core/legacy-widget' ],
+				// We can't transform if raw instance isn't shown in the REST API.
+				isMatch: ( { idBase, instance } ) =>
+					idBase === 'woocommerce_price_filter' && !! instance?.raw,
+				transform: ( { instance } ) =>
+					createBlock( 'woocommerce/price-filter', {
+						showInputFields: true,
+						showFilterButton: false,
```

Comment:
In order to better mimic how the widget works, I would set `showInputFields` to false and `showFilterButton` to true.